EIPS/eip-7906.md

```diff
@@ -0,0 +1,272 @@
+---
+eip: 7906
+title: Restricted Behavior Transaction Type
+description: A transaction type that provides a way for senders to restrict the outcomes of their execution
+author: Alex Forshtat (@forshtat), Shahaf Nacson (@shahafn), Dror Tirosh (@drortirosh)
+discussions-to: https://ethereum-magicians.org/t/eip-restricted-behavior-transaction-type/23130
+status: Draft
+type: Standards Track
+category: Core
+created: 2025-02-21
+---
+
+## Abstract
+
+This proposal defines the transaction payload to define write protections for balances and storage slots that are be enforced on the consensus level.
+These protected transactions empower Ethereum users to restrict the behavior of the deployed on-chain smart contracts without needing to perform a deep analysis of the EVM code.
+
+## Motivation
+
+The total value of crypto assets that have been stolen to date allegedly exceeds the yearly GDP of a medium-sized nation.
+This level of loss and waste is indefensible and has a long list of negative consequences for everyone involved.
+
+Specifically, one of the major ways in which Ethereum users end up losing their funds is not exposing private keys,
+but instead being tricked into signing transactions that appear innocent in their Wallet applications but in practice
+perform a complete sweep of the senders' assets.
+
+This situation is largely a result of two factors and must be addressed urgently.
+
+First, the ability of an average user to find, review, and analyze the code the transaction will execute is very limited.
+Second, the ability of a Wallet application to autonomously and reliably analyze this information is even more limited.
+
+This means that users have no mechanism to enforce any limits on what transactions will actually do once they are signed,
+and leaves a vast majority of users forced to perform blind signing in order to interact with Ethereum,
+exposing them to significant and avoidable risks.
+
+Existing transaction types in Ethereum only define their inputs, while the behavior and the outcome of
+transactions is only defined by the code permanently deployed on-chain.
+With the growing complexity of the deployed codebases and the growing monetary value controlled by this code,
+such an approach becomes increasingly more dangerous.
+
+By providing the Wallets and dApps with the ability to restrict the possible **outcomes** of a transaction,
+we can create a tool that users and wallets will apply to reduce their risk levels.
+
+## Specification
+
+### New Transaction Type
+
+We introduce a new [EIP-2718](./eip-2718.md) transaction, "restricted execution transaction", where the `TransactionType` is `RESTRICTED_EXECUTION_TX_TYPE` and the `TransactionPayload` is the RLP serialization of the following:
+
+```
+rlp([chain_id, nonce, max_priority_fee_per_gas, max_fee_per_gas, gas_limit, destination, value, data, access_list, authorization_list, execution_restrictions, signature_y_parity, signature_r, signature_s])
```

Comment:
What is `authorization_list`?

Reply:
The EIP-7702 authorization tuples. I followed the naming from the EIP-7702 itself:

```
The authorization_list is a list of tuples that store the address to code which the signer desires to execute in the context of their EOA. The transaction is considered invalid if the length of authorization_list is zero.
```

Should I remove the mention of `authorization_list` from this EIP or provide a link and an explanation of what this value means? I assumed that as EIP-7702 will be part of Ethereum very soon there is no need to re-explain it or "require" it specifically, am I wrong here?

Reply:
i think `require: 7702` in header sorts of resolve the dependency specification and terminolgy assumption

Reply:
please require all the transaction types EIPs you are trying to addess